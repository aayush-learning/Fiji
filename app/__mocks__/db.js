const usersDB = {
  sync: () => {
    return {rows: []}
  },
  allDocs: () => {
    return {rows: []}
  }
}

export { usersDB }