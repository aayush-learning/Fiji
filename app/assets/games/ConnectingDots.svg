<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 649.35 443.04"><defs><style>.cls-1{fill:#fafafa;}.cls-2{fill:#e37b40;}.cls-3{fill:#ffe0b2;}.cls-4{fill:#fff;}.cls-5{fill:#8d032a;}.cls-6{fill:#ffa726;}.cls-7{fill:#fff176;}</style></defs><title>Connecting Dots</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_2-2" data-name="Layer 2"><rect class="cls-1" width="648.95" height="443.04" rx="81.12" ry="81.12"/><path class="cls-2" d="M649,81.12V99.84H0V81.12A81.12,81.12,0,0,1,81.12,0H567.83A81.12,81.12,0,0,1,649,81.12Z"/><circle class="cls-3" cx="440.42" cy="51.29" r="36.37"/><circle cx="439.99" cy="50.44" r="20.82"/><circle class="cls-4" cx="442.96" cy="42.09" r="7.84"/><circle class="cls-3" cx="221.92" cy="51.29" r="36.37"/><circle cx="221.92" cy="50.08" r="20.82"/><circle class="cls-4" cx="224.5" cy="43.08" r="7.84"/><path class="cls-5" d="M76.21,0V210l-75-15-1-95C-1.9,45.3,19.29,8.73,76.21,0Z"/><path class="cls-5" d="M573.21,0V210l75-15,1-95C651.31,45.3,630.12,8.73,573.21,0Z"/><rect class="cls-6" x="86.63" y="193.83" width="93.36" height="93.36" rx="11.35" ry="11.35"/><rect class="cls-6" x="138.49" y="235.32" width="351.71" height="12.97"/><rect class="cls-7" x="141.49" y="369.32" width="351.71" height="12.97"/><rect class="cls-6" x="216.29" y="193.83" width="93.36" height="93.36" rx="11.35" ry="11.35"/><rect class="cls-6" x="343.37" y="193.83" width="93.36" height="93.36" rx="11.35" ry="11.35"/><rect class="cls-7" x="467.85" y="193.83" width="93.36" height="93.36" rx="11.35" ry="11.35"/><rect class="cls-7" x="86.63" y="326.09" width="93.36" height="93.36" rx="11.35" ry="11.35"/><rect class="cls-7" x="216.29" y="326.09" width="93.36" height="93.36" rx="11.35" ry="11.35"/><rect class="cls-7" x="343.37" y="326.09" width="93.36" height="93.36" rx="11.35" ry="11.35"/><rect class="cls-7" x="467.85" y="326.09" width="93.36" height="93.36" rx="11.35" ry="11.35"/></g></g></svg>