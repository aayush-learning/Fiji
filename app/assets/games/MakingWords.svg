<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 651 616.3"><defs><style>.cls-1{fill:#fff176;}.cls-2{fill:#ffb300;}.cls-3{fill:#fafafa;}.cls-4{fill:#ffe0b2;}.cls-5{fill:#fff;}.cls-6{fill:#fecb00;}</style></defs><title>Making Words</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_3" data-name="Layer 3"><polygon class="cls-1" points="622.1 233.37 465.68 233.37 621.9 2.01 622.1 233.37"/><polygon class="cls-2" points="621.9 319.38 535.02 233.37 621.9 103.12 621.9 319.38"/><polygon class="cls-1" points="187.27 233.37 29.18 0 29.18 233.37 187.27 233.37"/><polygon class="cls-2" points="29.18 319.38 117.39 233.37 29.18 103.12 29.18 319.38"/><path class="cls-3" d="M81.38,171.87H569.63A81.37,81.37,0,0,1,651,253.24V534.93a81.37,81.37,0,0,1-81.37,81.37H81.37A81.37,81.37,0,0,1,0,534.93V253.24a81.38,81.38,0,0,1,81.38-81.37Z"/><path class="cls-1" d="M651,253.24V272H0V253.24a81.37,81.37,0,0,1,81.37-81.37H569.62A81.37,81.37,0,0,1,651,253.24Z"/><circle class="cls-4" cx="441.81" cy="223.32" r="36.48"/><circle cx="441.38" cy="222.47" r="20.88"/><circle class="cls-5" cx="444.36" cy="214.09" r="7.87"/><circle class="cls-4" cx="222.62" cy="223.32" r="36.48"/><circle cx="222.62" cy="222.1" r="20.88"/><circle class="cls-5" cx="225.21" cy="215.09" r="7.87"/><rect class="cls-1" x="182.09" y="444.57" width="13.66" height="113.76" transform="translate(-194.17 126.77) rotate(-24.99)"/><rect class="cls-1" x="328.75" y="397.26" width="13.66" height="113.76" transform="translate(-160.45 184.3) rotate(-24.99)"/><rect class="cls-1" x="232.62" y="531.15" width="113.76" height="13.66" transform="translate(-200.18 172.68) rotate(-24.99)"/><circle class="cls-6" cx="161.92" cy="433.5" r="43.14"/><circle class="cls-6" cx="221.61" cy="567.82" r="43.14"/><rect class="cls-1" x="466.86" y="332.17" width="13.66" height="113.76" transform="translate(-120.41 238.09) rotate(-25.14)"/><rect class="cls-1" x="316.13" y="345.95" width="113.76" height="13.66" transform="translate(-114.54 191.88) rotate(-25.14)"/><circle class="cls-6" cx="500.86" cy="456.93" r="43.14"/><circle class="cls-6" cx="440.83" cy="322.77" r="43.14"/><circle class="cls-6" cx="360.38" cy="510.01" r="43.14"/><circle class="cls-6" cx="305.88" cy="390.36" r="43.14"/></g></g></svg>