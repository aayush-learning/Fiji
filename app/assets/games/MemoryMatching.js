export default `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 650.99 521.82"><defs><style>.cls-1{fill:#ff80ab;}.cls-2,.cls-3{fill:#e0dcdd;}.cls-3{font-size:72.28px;font-family:ArialMT, Arial;}.cls-4{fill:#efe8eb;}</style></defs><title>Memory Matching</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_2-2" data-name="Layer 2"><circle class="cls-1" cx="544.03" cy="67.86" r="67.86"/><circle class="cls-2" cx="538.82" cy="85.09" r="42.64"/><circle class="cls-1" cx="119.73" cy="67.86" r="67.86"/><circle class="cls-2" cx="124.94" cy="85.09" r="42.64"/><rect class="cls-2" y="77.04" width="650.99" height="444.78" rx="81.32" ry="81.32"/><rect class="cls-1" x="172.5" y="283.87" width="30.82" height="11.34"/><rect class="cls-1" x="315.05" y="283.87" width="30.82" height="11.34"/><rect class="cls-1" x="457.6" y="283.87" width="30.82" height="11.34"/><path class="cls-1" d="M167.17,348.71h-95a12.81,12.81,0,0,1-12.77-12.78v-95a12.81,12.81,0,0,1,12.77-12.78h95A12.82,12.82,0,0,1,180,240.91v95A12.82,12.82,0,0,1,167.17,348.71Z"/><path class="cls-1" d="M448.38,348.71h-95a12.81,12.81,0,0,1-12.77-12.78v-95a12.81,12.81,0,0,1,12.77-12.78h95a12.82,12.82,0,0,1,12.78,12.78v95A12.82,12.82,0,0,1,448.38,348.71Z"/><path class="cls-1" d="M306.43,348.71h-95a12.82,12.82,0,0,1-12.78-12.78v-95a12.82,12.82,0,0,1,12.78-12.78h95a12.81,12.81,0,0,1,12.77,12.78v95A12.81,12.81,0,0,1,306.43,348.71Z"/><path class="cls-1" d="M589,345.45H494a12.81,12.81,0,0,1-12.77-12.78v-95A12.81,12.81,0,0,1,494,224.88h95a12.81,12.81,0,0,1,12.78,12.77v95A12.82,12.82,0,0,1,589,345.45Z"/><text class="cls-3" transform="translate(382.13 313.23)">?</text><text class="cls-3" transform="translate(94.25 313.23)">?</text><polygon class="cls-2" points="544.52 252.03 554.47 272.17 576.7 275.4 560.61 291.08 564.41 313.23 544.52 302.77 524.64 313.23 528.44 291.08 512.35 275.4 534.58 272.17 544.52 252.03"/><polygon class="cls-2" points="256.21 254.56 266.15 274.71 288.38 277.94 272.29 293.62 276.09 315.76 256.21 305.31 236.32 315.76 240.12 293.62 224.03 277.94 246.26 274.71 256.21 254.56"/><rect class="cls-1" x="172.37" y="423.51" width="30.82" height="11.34"/><rect class="cls-1" x="314.92" y="423.51" width="30.82" height="11.34"/><rect class="cls-1" x="457.48" y="423.51" width="30.82" height="11.34"/><path class="cls-1" d="M167,488.35H72a12.82,12.82,0,0,1-12.78-12.78v-95A12.81,12.81,0,0,1,72,367.78h95a12.81,12.81,0,0,1,12.78,12.77v95A12.82,12.82,0,0,1,167,488.35Z"/><path class="cls-1" d="M448.25,488.35h-95a12.82,12.82,0,0,1-12.78-12.78v-95a12.81,12.81,0,0,1,12.78-12.77h95A12.81,12.81,0,0,1,461,380.55v95A12.82,12.82,0,0,1,448.25,488.35Z"/><path class="cls-1" d="M306.3,488.35h-95a12.82,12.82,0,0,1-12.78-12.78v-95a12.81,12.81,0,0,1,12.78-12.77h95a12.81,12.81,0,0,1,12.77,12.77v95A12.81,12.81,0,0,1,306.3,488.35Z"/><path class="cls-1" d="M588.89,485.09h-95a12.81,12.81,0,0,1-12.77-12.78v-95a12.81,12.81,0,0,1,12.77-12.77h95a12.81,12.81,0,0,1,12.78,12.77v95A12.82,12.82,0,0,1,588.89,485.09Z"/><text class="cls-3" transform="translate(382.01 452.87)">?</text><text class="cls-3" transform="translate(94.12 452.87)">?</text><polygon class="cls-2" points="544.4 391.67 554.34 411.82 576.57 415.05 560.49 430.73 564.28 452.87 544.4 442.42 524.51 452.87 528.31 430.73 512.22 415.05 534.46 411.82 544.4 391.67"/><polygon class="cls-2" points="256.08 394.2 266.02 414.35 288.25 417.58 272.17 433.26 275.96 455.4 256.08 444.95 236.19 455.4 239.99 433.26 223.9 417.58 246.14 414.35 256.08 394.2"/><path class="cls-1" d="M569.67,77H81.32A81.32,81.32,0,0,0,0,158.36v41.1H651v-41.1A81.32,81.32,0,0,0,569.67,77Z"/><circle class="cls-2" cx="449.81" cy="142.08" r="41.38"/><circle cx="449.32" cy="141.12" r="23.69"/><circle class="cls-4" cx="458.73" cy="126.36" r="8.92"/><circle class="cls-2" cx="201.18" cy="142.08" r="41.38"/><circle cx="201.18" cy="140.71" r="23.69"/><circle class="cls-4" cx="210.1" cy="126.36" r="8.92"/></g></g></svg>`