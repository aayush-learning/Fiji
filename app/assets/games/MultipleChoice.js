export default `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 650 603.69"><defs><style>.cls-1{fill:#ffb300;}.cls-2{fill:#ff8a65;}.cls-3{fill:#fafafa;}.cls-4{fill:#ffe0b2;}.cls-5{fill:#fff;}.cls-6{fill:none;stroke:#000;stroke-miterlimit:10;stroke-width:15px;}.cls-7{fill:#1ab5a2;}.cls-8{fill:#ff525f;}.cls-9{fill:#ff7c06;}</style></defs><title>Multiple Choice</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_3" data-name="Layer 3"><polygon class="cls-1" points="621.57 233.01 463.72 233.01 619.7 2 621.57 233.01"/><polygon class="cls-2" points="619.7 318.89 532.95 233.01 619.7 102.97 619.7 318.89"/><polygon class="cls-1" points="185.73 233.01 27.89 0 27.89 233.01 185.73 233.01"/><polygon class="cls-2" points="27.89 318.89 115.97 233.01 27.89 102.96 27.89 318.89"/><rect class="cls-3" y="159.94" width="650" height="443.75" rx="81.25" ry="81.25"/><path class="cls-1" d="M650,241.19v18.75H0V241.19a81.25,81.25,0,0,1,81.25-81.25h487.5A81.25,81.25,0,0,1,650,241.19Z"/><circle class="cls-4" cx="441.13" cy="211.31" r="36.42"/><circle cx="440.7" cy="210.47" r="20.85"/><circle class="cls-5" cx="443.67" cy="202.1" r="7.85"/><circle class="cls-4" cx="222.27" cy="211.31" r="36.42"/><circle cx="222.27" cy="210.1" r="20.85"/><circle class="cls-5" cx="224.86" cy="203.09" r="7.85"/><polyline class="cls-6" points="279.67 344.2 155.69 344.2 155.69 465.16"/><polyline class="cls-6" points="370.39 344.2 494.38 344.2 494.38 465.16"/><path class="cls-7" d="M376.94,418.28H263A15.36,15.36,0,0,1,247.67,403V288.82A15.37,15.37,0,0,1,263,273.5h114a15.37,15.37,0,0,1,15.32,15.32v114A15.41,15.41,0,0,1,376.94,418.28Z"/><path class="cls-8" d="M218.79,595.19H104.84a15.36,15.36,0,0,1-15.32-15.32V465.74a15.36,15.36,0,0,1,15.32-15.32h114a15.37,15.37,0,0,1,15.32,15.32V579.68A15.41,15.41,0,0,1,218.79,595.19Z"/><path class="cls-9" d="M547.18,593.66H433.23a15.37,15.37,0,0,1-15.32-15.32V464.4a15.37,15.37,0,0,1,15.32-15.33H547.18A15.37,15.37,0,0,1,562.5,464.4V578.34A15.36,15.36,0,0,1,547.18,593.66Z"/></g></g></svg>`