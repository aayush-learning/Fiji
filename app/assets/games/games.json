{
  "game:memory-matching": {
    "_id": "game:memory-matching",
    "name": "Memory Matching",
    "sets": [
      {
        "_id": "letters",
        "name": "Letters"
      },
      {
        "_id": "numbers",
        "name": "Numbers"
      },
      {
        "_id": "images",
        "name": "Images"
      }
    ]
  },
  "game:tap-home": {
    "_id": "game:tap-home",
    "name": "Tap Home",
    "sets": [
      {
        "_id": "letters",
        "name": "Letters"
      },
      {
        "_id": "numbers",
        "name": "Numbers"
      },
      {
        "_id": "images",
        "name": "Images"
      }
    ]
  },
  "game:reflex": {
    "_id": "game:reflex",
    "name": "Reflex",
    "sets": [
      {
        "_id": "letters",
        "name": "Letters"
      },
      {
        "_id": "numbers",
        "name": "Numbers"
      },
      {
        "_id": "images",
        "name": "Images"
      }
    ]
  },
  "game:tap-wrong": {
    "_id": "game:tap-wrong",
    "name": "Tap-wrong",
    "sets": [
      {
        "_id": "letters",
        "name": "Letters"
      },
      {
        "_id": "numbers",
        "name": "Numbers"
      },
      {
        "_id": "images",
        "name": "Images"
      }
    ]
  },
  "game:word": {
    "_id": "game:word",
    "name": "Word",
    "sets": [
      {
        "_id": "letters",
        "name": "Letters"
      },
      {
        "_id": "numbers",
        "name": "Numbers"
      },
      {
        "_id": "images",
        "name": "Images"
      }
    ]
  },
  "game:multiple-choice": {
    "_id": "game:multiple-choice",
    "name": "Multiple-choice",
    "sets": [
      {
        "_id": "letters",
        "name": "Letters"
      },
      {
        "_id": "numbers",
        "name": "Numbers"
      },
      {
        "_id": "images",
        "name": "Images"
      }
    ]
  },
  "game:connect-dots": {
    "_id": "game:connect-dots",
    "name": "Connect-dots",
    "sets": [
      {
        "_id": "letters",
        "name": "Letters"
      },
      {
        "_id": "numbers",
        "name": "Numbers"
      },
      {
        "_id": "images",
        "name": "Images"
      }
    ]
  }
}