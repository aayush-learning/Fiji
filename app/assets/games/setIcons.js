const setIcons = {
  'missing': require('./missing.js'),
  'letters': require('./letters.js'),
  'numbers': require('./numbers.js'),
  'images': require('./images.js'),
  'game:memory-matching': require('./MemoryMatching.js'),
  'game:tap-home': require('./TapNext.js'),
  'game:reflex': require('./letters.js'),
  'game:tap-wrong': require('./TapWrong.js'),
  'game:word': require('./MakingWords.js'),
  'game:multiple-choice': require('./MultipleChoice.js'),
  'game:connect-dots': require('./ConnectingDots.js')
}

export default setIcons