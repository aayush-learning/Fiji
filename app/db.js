import PouchDB from 'pouchdb-react-native';

// these settings let us use cuids
// see: https://github.com/ericelliott/cuid/issues/54
global.navigator.mimeTypes = '';
global.navigator.userAgent = 'reactnative';    

const fijiDB = new PouchDB('fiji', { adapter: 'asyncstorage' });
console.log(fijiDB.adapter)
fijiDB.info().then(function (info) {
  console.log(info);
})
export default fijiDB