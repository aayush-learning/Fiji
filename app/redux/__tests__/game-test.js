jest.mock('../../db')

import { contentDB } from '../../db'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import games, { initialState, fetchGamesFailure, fetchGamesRequest, fetchGamesSuccess, fetchGames } from '../game'
import { games as gamesData } from '../../../config/jest/mockData'

// test reducers

it('returns the same state on an unhandled action', () => {
  expect(games(initialState, {type: '_NULL'})).toMatchSnapshot()
})

it('handles FETCH_GAMES_REQUEST action', () => {
  expect(games(initialState, fetchGamesRequest())).toMatchSnapshot()
})

it('handles FETCH_GAMES_SUCCESS action', () => {
  expect(games(initialState, fetchGamesSuccess(gamesData))).toMatchSnapshot()
})

it('handles FETCH_GAMES_FAILURE action', () => {
  expect(games(initialState, fetchGamesFailure())).toMatchSnapshot()
})

// test actions

it('creates a FETCH_GAMES_REQUEST action', () => {
  expect(fetchGamesRequest()).toMatchSnapshot()
})

it('creates a FETCH_GAMES_SUCCESS action', () => {
  expect(fetchGamesSuccess(gamesData)).toMatchSnapshot()
})

it('creates a FETCH_GAMES_FAILURE action', () => {
  expect(fetchGamesFailure()).toMatchSnapshot()
})

// test thunks

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

it('handles fetchGames action', async () => {
  const store = mockStore(initialState)
  contentDB.allDocs = jest.fn()
    .mockReturnValue({
      rows: gamesData.map(function(doc) {
        return { doc }
      })
    })
  await store.dispatch(fetchGames())
  expect(store.getActions()).toMatchSnapshot()
})  