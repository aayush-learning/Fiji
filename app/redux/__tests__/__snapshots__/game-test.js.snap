// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a FETCH_GAMES_FAILURE action 1`] = `
Object {
  "type": "Fiji/game/FETCH_GAMES_FAILURE",
}
`;

exports[`creates a FETCH_GAMES_REQUEST action 1`] = `
Object {
  "type": "Fiji/game/FETCH_GAMES_REQUEST",
}
`;

exports[`creates a FETCH_GAMES_SUCCESS action 1`] = `
Object {
  "games": Array [
    Object {
      "_id": "game:xyz1",
      "category": "listening",
      "name": "Xyz1",
    },
    Object {
      "_id": "game:xyz2",
      "category": "listening",
      "name": "Xyz2",
    },
    Object {
      "_id": "game:xyz3",
      "category": "listening",
      "name": "Xyz3",
    },
    Object {
      "_id": "game:xyz4",
      "category": "listening",
      "name": "Xyz4",
    },
    Object {
      "_id": "game:xyz5",
      "category": "writing",
      "name": "Xyz5",
    },
    Object {
      "_id": "game:xyz6",
      "category": "writing",
      "name": "Xyz6",
    },
    Object {
      "_id": "game:xyz7",
      "category": "writing",
      "name": "Xyz7",
    },
    Object {
      "_id": "game:xyz8",
      "category": "writing",
      "name": "Xyz8",
    },
    Object {
      "_id": "game:alphamon",
      "category": "reading",
      "name": "Alphamon",
    },
    Object {
      "_id": "game:alphamon1",
      "category": "reading",
      "name": "Alphamon1",
    },
    Object {
      "_id": "game:alphamon2",
      "category": "reading",
      "name": "Alphamon2",
    },
    Object {
      "_id": "game:alphamon3",
      "category": "reading",
      "name": "Alphamon3",
    },
    Object {
      "_id": "game:connect-the-dots",
      "category": "maths",
      "name": "Connect The Dots",
    },
    Object {
      "_id": "game:connect-the-dot1",
      "category": "maths",
      "name": "Connect The Dots1",
    },
    Object {
      "_id": "game:grocery-store",
      "category": "maths",
      "name": "Grocery Store",
    },
    Object {
      "_id": "game:grocery-store2",
      "category": "maths",
      "name": "Grocery Store2",
    },
  ],
  "type": "Fiji/game/FETCH_GAMES_SUCCESS",
}
`;

exports[`handles FETCH_GAMES_FAILURE action 1`] = `
Object {
  "games": Array [],
  "isFetching": false,
}
`;

exports[`handles FETCH_GAMES_REQUEST action 1`] = `
Object {
  "games": Array [],
  "isFetching": true,
}
`;

exports[`handles FETCH_GAMES_SUCCESS action 1`] = `
Object {
  "games": Array [
    Object {
      "_id": "game:xyz1",
      "category": "listening",
      "name": "Xyz1",
    },
    Object {
      "_id": "game:xyz2",
      "category": "listening",
      "name": "Xyz2",
    },
    Object {
      "_id": "game:xyz3",
      "category": "listening",
      "name": "Xyz3",
    },
    Object {
      "_id": "game:xyz4",
      "category": "listening",
      "name": "Xyz4",
    },
    Object {
      "_id": "game:xyz5",
      "category": "writing",
      "name": "Xyz5",
    },
    Object {
      "_id": "game:xyz6",
      "category": "writing",
      "name": "Xyz6",
    },
    Object {
      "_id": "game:xyz7",
      "category": "writing",
      "name": "Xyz7",
    },
    Object {
      "_id": "game:xyz8",
      "category": "writing",
      "name": "Xyz8",
    },
    Object {
      "_id": "game:alphamon",
      "category": "reading",
      "name": "Alphamon",
    },
    Object {
      "_id": "game:alphamon1",
      "category": "reading",
      "name": "Alphamon1",
    },
    Object {
      "_id": "game:alphamon2",
      "category": "reading",
      "name": "Alphamon2",
    },
    Object {
      "_id": "game:alphamon3",
      "category": "reading",
      "name": "Alphamon3",
    },
    Object {
      "_id": "game:connect-the-dots",
      "category": "maths",
      "name": "Connect The Dots",
    },
    Object {
      "_id": "game:connect-the-dot1",
      "category": "maths",
      "name": "Connect The Dots1",
    },
    Object {
      "_id": "game:grocery-store",
      "category": "maths",
      "name": "Grocery Store",
    },
    Object {
      "_id": "game:grocery-store2",
      "category": "maths",
      "name": "Grocery Store2",
    },
  ],
  "isFetching": false,
}
`;

exports[`handles fetchGames action 1`] = `
Array [
  Object {
    "type": "Fiji/game/FETCH_GAMES_REQUEST",
  },
  Object {
    "games": Array [
      Object {
        "_id": "listening",
        "category": "listening",
        "games": Array [
          Object {
            "_id": "game:xyz1",
            "category": "listening",
            "name": "Xyz1",
          },
          Object {
            "_id": "game:xyz2",
            "category": "listening",
            "name": "Xyz2",
          },
          Object {
            "_id": "game:xyz3",
            "category": "listening",
            "name": "Xyz3",
          },
          Object {
            "_id": "game:xyz4",
            "category": "listening",
            "name": "Xyz4",
          },
        ],
      },
      Object {
        "_id": "writing",
        "category": "writing",
        "games": Array [
          Object {
            "_id": "game:xyz5",
            "category": "writing",
            "name": "Xyz5",
          },
          Object {
            "_id": "game:xyz6",
            "category": "writing",
            "name": "Xyz6",
          },
          Object {
            "_id": "game:xyz7",
            "category": "writing",
            "name": "Xyz7",
          },
          Object {
            "_id": "game:xyz8",
            "category": "writing",
            "name": "Xyz8",
          },
        ],
      },
      Object {
        "_id": "reading",
        "category": "reading",
        "games": Array [
          Object {
            "_id": "game:alphamon",
            "category": "reading",
            "name": "Alphamon",
          },
          Object {
            "_id": "game:alphamon1",
            "category": "reading",
            "name": "Alphamon1",
          },
          Object {
            "_id": "game:alphamon2",
            "category": "reading",
            "name": "Alphamon2",
          },
          Object {
            "_id": "game:alphamon3",
            "category": "reading",
            "name": "Alphamon3",
          },
        ],
      },
      Object {
        "_id": "maths",
        "category": "maths",
        "games": Array [
          Object {
            "_id": "game:connect-the-dots",
            "category": "maths",
            "name": "Connect The Dots",
          },
          Object {
            "_id": "game:connect-the-dot1",
            "category": "maths",
            "name": "Connect The Dots1",
          },
          Object {
            "_id": "game:grocery-store",
            "category": "maths",
            "name": "Grocery Store",
          },
          Object {
            "_id": "game:grocery-store2",
            "category": "maths",
            "name": "Grocery Store2",
          },
        ],
      },
    ],
    "type": "Fiji/game/FETCH_GAMES_SUCCESS",
  },
]
`;

exports[`returns the same state on an unhandled action 1`] = `
Object {
  "games": Array [],
  "isFetching": false,
}
`;
